src/internal/logger.hh

```diff
@@ -0,0 +1,38 @@
+#include "zarr.h"
```

Comment:
This design seems fragile. We're incorporating a C API wrapper into a logger class that should be generalized for accessing properties—properties that ought to be defined here initially. Are there any alternative approaches we could consider?